data-protocols/dsp/dsp-http-core/src/main/java/org/eclipse/edc/protocol/dsp/util/ErrorUtil.java

```diff
@@ -0,0 +1,90 @@
+/*
+ *  Copyright (c) 2023 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.protocol.dsp.util;
+
+import jakarta.json.Json;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.eclipse.edc.jsonld.spi.JsonLdKeywords;
+import org.eclipse.edc.web.spi.exception.AuthenticationFailedException;
+import org.eclipse.edc.web.spi.exception.BadGatewayException;
+import org.eclipse.edc.web.spi.exception.InvalidRequestException;
+import org.eclipse.edc.web.spi.exception.NotAuthorizedException;
+import org.eclipse.edc.web.spi.exception.ObjectConflictException;
+import org.eclipse.edc.web.spi.exception.ObjectNotFoundException;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static jakarta.ws.rs.core.Response.Status.BAD_GATEWAY;
+import static jakarta.ws.rs.core.Response.Status.BAD_REQUEST;
+import static jakarta.ws.rs.core.Response.Status.CONFLICT;
+import static jakarta.ws.rs.core.Response.Status.FORBIDDEN;
+import static jakarta.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+import static jakarta.ws.rs.core.Response.Status.NOT_FOUND;
+import static jakarta.ws.rs.core.Response.Status.NOT_IMPLEMENTED;
+import static jakarta.ws.rs.core.Response.Status.UNAUTHORIZED;
+import static org.eclipse.edc.jsonld.spi.Namespaces.DSPACE_SCHEMA;
+
+
+/**
+ * Creates CatalogError, ContractNegotiation and TransferProcessError Responses for the Dataspace Protocol Apis
+ * {DspNegotiationController}, {CatalogController} and {DspTransferProcessApiController}
+ *
+ */
+public class ErrorUtil {
+
+    /** Creating ErrorResponse
+     *
+     * @param type dspace:ContractNegotiationError/dspace:TransferProcessError/dspace:CatalogError depending on executing Controller
+     * @param processId referenced processId Error is occurring
+     * @param throwable describing ErrorMessage
+     * @return {@link Response} ErrorResponse
+     */
+    public static Response createErrorResponse(String type, Optional<String> processId, Throwable throwable) {
```

Comment:
to follow an approach more similar to the existing, we could introduce three pojos:
- `CatalogErrorMessage`
- `ContractNegotiationErrorMessage`
- `TransferProcessErrorMessage`

each one containing a field `processId` and a field with the throwable.
and for each one implement a `*ErrorToResponse` that will take care of creating the `Response` with the `JsonObject`. 
this will simplify tests, separate responsibilities, avoid the util class and remove unneeded direct dependencies to the `dsp-http-core` module.